recipe-server/Dockerfile

```diff
@@ -10,17 +10,17 @@ RUN apt-get update && \
 RUN curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - && \
     echo 'deb https://deb.nodesource.com/node_8.x jessie main' > /etc/apt/sources.list.d/nodesource.list && \
     echo 'deb-src https://deb.nodesource.com/node_8.x jessie main' >> /etc/apt/sources.list.d/nodesource.list && \
-    apt-get update && apt-get install -y nodejs
-RUN npm install -g npm@5
+    apt-get update && apt-get install -y nodejs && \
+    curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    echo 'deb https://dl.yarnpkg.com/debian/ stable main' > /etc/apt/sources.list.d/yarn.list && \
+    apt-get update && apt-get install yarn
 
-COPY ./requirements /app/requirements
-COPY ./package.json /app/package.json
+COPY . /app
```

Comment:
Moving app up this far will make the Docker layer cache miss all future steps most of the time, making working with Docker locally a lot slower (though it doesn't affect CircleCI). I think it would be better to continue copying only the things needed (`requirements/`, `package.json`, `package-lock.json`).

Do we need to keep the vendor post-install step? It can be convenient, but I think it mostly gets in my way personally. Alternatively, we could also include the webpack vendor file in this copy.

That being said, since we rarely do local development using Docker, the layer cache isn't actually saving us much work. If this is hard to fix, we can leave it as it is now. It is certainly simpler.

Reply:
> Do we need to keep the vendor post-install step?

Not really. I had envisioned it being useful as we add new packages, but it indeed has become more of a hassle than anything. I've removed it from the `package.json` and reverted the `COPY` stuff in the Dockerfile. I also removed an errant `yarn` dependency I found in the `package.json`.